core/src/main/java/com/linecorp/armeria/common/stream/DeferredStreamMessage.java

```diff
@@ -366,32 +367,63 @@ public final void abort(Throwable cause) {
             return;
         }
 
-        final SubscriptionImpl newSubscription = new SubscriptionImpl(
-                this, AbortingSubscriber.get(cause), ImmediateEventExecutor.INSTANCE, EMPTY_OPTIONS);
-        downstreamSubscriptionUpdater.compareAndSet(this, null, newSubscription);
+        if (!subscribedToUpstreamUpdater.compareAndSet(this, 0, 1)) {
```

Comment:
Q) It seems a `Subscriber` that fails to update `subscribedToUpstream` to 1 at L349 does not receive additional events after `onSubscribe()`. Should we call `onError(new IllegalStateException("subscribed by other subscriber already"))`?

Reply:
Ah, that is correct. I forgot to add it. 😓

Reply:
The downstream is actually aborted by this line:
https://github.com/line/armeria/blob/5050f87024a809cde375c1c8227de462b6e6789d/core/src/main/java/com/linecorp/armeria/common/stream/DeferredStreamMessage.java#L406

Reply:
Good point. That's right. 
By the way, while I was checking your code point, I realized that `onError()` L418 may be called before `onSubscribe()` L332.

Reply:
That is correct. 😓 Thanks a lot for finding this. 😆